### PR TITLE
PI-999 - Implemented solution to prevent Playwright report from exposing auth tokens in case of test failures

### PR DESCRIPTION
**Pull request by smalepati99**
This Pull Request addresses the issue with Playwright reports leaking auth tokens on failure. To resolve this issue, we have added a **sanitiseError** function that takes any async function as an argument and wraps it with a try-catch block. This function catches any error thrown by the passed function and logs it to the console, preventing any sensitive information from being leaked. In addition, we have also defined a type for async functions to ensure better type safety. This solution ensures that we can avoid any further security issues with the Playwright report while keeping the code simple and maintainable.

In addition, we have made changes to the functions in **prison-api.ts**, such as the **createPrisoner** function, by incorporating the sanitiseError function to enclose them within a try-catch block. As a result, the functions now export as sanitised error functions and provide a custom error message in case an error occurs during execution.

We have also made changes to the tests by removing the expect(response.ok()).toBeTruthy() assertion and adding the failOnStatusCode: true option to the request in order to catch any HTTP errors and handle them appropriately. This ensures that any failed requests are properly handled and do not leak any sensitive information.

### CODE REVIEW

File: tests/prison-custody-status-to-delius/release-recall-test.spec.ts [hunk 1]
```diff
@@ -57,7 +57,7 @@ test('Release and recall test', async ({ page }) => {
 
 test('Temporary absence test', async ({ page }) => {
     test.slow() // increase the timeout - releases/recall publishing can take a few minutes
-
+    page.on('console', msg => console.log(msg.text()))
```
Comment (marcus-bcl) [inline tests/prison-custody-status-to-delius/release-recall-test.spec.ts]: You'll need to remove this, otherwise we'll still log the auth token to the console

File: steps/workforce/utils.ts [hunk 2]
```diff
@@ -70,7 +70,9 @@ export const allocateUnallocatedCasesWithinDateRange = async (
 
             // Allocate to team/staff
             await page
-                .locator('tr', { hasText: `${allocation.staff.firstName} ${allocation.staff.lastName}` })
+                // .locator('tr', { hasText: `${allocation.staff.firstName} ${allocation.staff.lastName}` })
+                .locator('tr', { hasText: `Isaiah Lakanu` })
+
```
Comment (marcus-bcl) [inline steps/workforce/utils.ts]: Is this change intentional?

**Review by marcus-bcl — CHANGES_REQUESTED** [review]
Looks good 🤖

File: tests/prison-custody-status-to-delius/release-recall-test.spec.ts (hunk 1 above)
Comment (smalepati99) [reply tests/prison-custody-status-to-delius/release-recall-test.spec.ts]: I will remove it straight away. Somehow missed taking it off.

File: steps/workforce/utils.ts (hunk 2 above)
Comment (smalepati99) [reply steps/workforce/utils.ts]: I was testing workforce. I have now reverted back, thanks a lot for the comment.

File: tests/prison-custody-status-to-delius/release-recall-test.spec.ts (hunk 1 above)
Comment (smalepati99) [reply tests/prison-custody-status-to-delius/release-recall-test.spec.ts]: Removed now.

**Review by marcus-bcl — APPROVED** [review]
